src/matcha_ml/services/azure_service.py

```diff
@@ -172,6 +173,53 @@ def fetch_resource_groups(self) -> Dict[str, ResourceGroup]:
 
             return self._resource_groups
 
+    def fetch_storage_access_key(
+        self, resource_group_name: str, storage_account_name: str
+    ) -> str:
+        """Fetches one of the storage account's access key from Azure and returns it.
+
+        Args:
+            resource_group_name (str): Name of resource group
+            storage_account_name (str): Name of storage account
+
+        Returns:
+            str: One of the acccess key corresponding to storage account
+        """
+        self._storage_client = StorageManagementClient(
+            self._credential, str(self.subscription_id)
+        )
+        keys = self._storage_client.storage_accounts.list_keys(
```

Comment:
Just want to confirm, this will only work when user is login to Azure?

Reply:
yes, there's a try and except block to catch and raise exception if any error arises.